File: app/src/main/java/sk/styk/martin/apkanalyzer/adapter/detaillist/SimpleStringListAdapter.java
package sk.styk.martin.apkanalyzer.adapter.detaillist;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import sk.styk.martin.apkanalyzer.R;

/**
 * Created by Martin Styk on 07.07.2017.
 */
public class SimpleStringListAdapter extends GenericDetailListAdapter<String, SimpleStringListAdapter.ViewHolder> {

    public SimpleStringListAdapter(@NonNull List<String> items) {
        super(items);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_simple_string, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        holder.name.setText(getItem(position));
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        final TextView name;

        ViewHolder(View v) {
            super(v);
            name = v.findViewById(R.id.item_detail_string);
        }
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/adapter/detaillist/SimpleStringListAdapter.kt
package sk.styk.martin.apkanalyzer.adapter.detaillist

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import sk.styk.martin.apkanalyzer.databinding.ListItemSimpleStringBinding

/**
 * @author Martin Styk
 * @version 07.07.2017.
 */
class SimpleStringListAdapter(items: List<String>) : GenericDetailListAdapter<String, SimpleStringListAdapter.ViewHolder>(items) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater: LayoutInflater = LayoutInflater.from(parent.context)
        val itemBinding = ListItemSimpleStringBinding.inflate(layoutInflater, parent, false)
        return ViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }

    inner class ViewHolder(val binding: ListItemSimpleStringBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: String) {
            binding.value = item
            binding.executePendingBindings()
        }
    }
}